src/chunk.c

```diff
@@ -2151,23 +2266,143 @@ ts_chunk_drop(Chunk *chunk, DropBehavior behavior, int32 log_level)
 			 chunk->fd.table_name.data);
 
 	/* Remove the chunk from the hypertable table */
-	ts_chunk_delete_by_relid(chunk->table_id, behavior);
+	ts_chunk_delete_by_relid(chunk->table_id, behavior, preserve_catalog_row);
```

Comment:
need a concurrent test where 1) 1 txn drops a chunk 2) the other txn is trying to recreate the chunk. this is to check that we are holding all locks till the first txn completes.

Reply:
The lock on line 742 `LockRelationOid(ht->main_table_relid, ShareUpdateExclusiveLock);` prevents this and hasn't changed with this PR so I don't think we need more tests here.